cabal-install/src/Distribution/Client/Config.hs

```diff
@@ -591,7 +591,22 @@ initialSavedConfig = do
     }
   }
 
--- | If @CABAL\_DIR@ is set or @~/.cabal@ exists, return that
+-- | Issue a warning if both @$XDG_CONFIG_HOME/cabal/config@ and
+-- @~/.cabal@ exists.
+warnOnTwoConfigs :: Verbosity -> IO ()
+warnOnTwoConfigs verbosity = do
+  defaultDir <- getAppUserDataDirectory "cabal"
+  dotCabalExists <- doesDirectoryExist defaultDir
+  xdgCfg <- getXdgDirectory XdgConfig ("cabal" </> "config")
+  xdgCfgExists <- doesFileExist xdgCfg
+  when (dotCabalExists && xdgCfgExists) $
+    warn verbosity $
+    "Both " <> defaultDir <>
+    " and " <> xdgCfg <>
+    " exist - ignoring the former."
+
+-- | If @CABAL\_DIR@ is set or @~/.cabal@ exists (and
```

Comment:
The parenthesis are confusing. How about:

> If @CABAL\_DIR@ is set return @Just@ its value. Otherwise, if @~/.cabal@ exists and
> @$XDG_CONFIG_HOME/cabal/config@ does not exist, return @Just "~/.cabal"@. 
> Otherwise, return @Nothing@.